packages/skeleton-react/src/lib/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { createContext, useCallback, useContext, useEffect, useRef, useState } from "react";
+import { IconProps, RatingContextState, RatingProps } from "./types";
+
+// Context
+
+export const RatingContext = createContext<RatingContextState>({
+    interactive: false,
+    value: 0,
+    order: 0,
+    figureRef: undefined
+});
+
+// Components ---
+const RatingRoot: React.FC<RatingProps> = ({
+    value = 0,
+    max = 5,
+    interactive = false,
+    fraction = 1,
+    // Root
+    base = 'flex',
+    width = 'w-full',
+    text = 'text-token',
+    fill = 'fill-token',
+    justify = 'justify-center',
+    spaceX = 'space-x-2',
+    classes = '',
+    // Item ---
+    buttonBase = 'w-full h-full',
+    buttonPosition = 'relative',
+    buttonAspect = 'aspect-square',
+    buttonClasses = '',
+    // Events
+    onMouseDown = () => {},
+    onKeyDown = () => {},
+    onValueChange = () => {},
+	// Children
+	children,
+}) => {
+    const figureRef = useRef<HTMLElement>(null);
+
+    const [focusedButtonIndex, setFocusedButtonIndex] = useState(0);
+    useEffect(() => {
+        const index = Math.max(0, Math.ceil(value - 1));
+        setFocusedButtonIndex(index);
+    }, [value]);
+
+    const onRatingMouseDown = useCallback((event: React.MouseEvent<HTMLButtonElement>, order: number) => {
+        if(!figureRef.current) return;
+
+        const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+        const fractionWidth = ratingRect.width / fraction;
+        const left = event.clientX - ratingRect.left;
+        let selectedFraction = Math.floor(left / fractionWidth) + 1;
+
+        if(getComputedStyle(figureRef.current).direction === 'rtl') {
+            selectedFraction = fraction - selectedFraction + 1;
+        }
+
+        value = order + selectedFraction / fraction;
+        onValueChange(value);
+        onMouseDown(event, value);
+    }, [fraction]);
+
+    const onRatingKeyDown = useCallback((event: React.KeyboardEvent<HTMLButtonElement>) => {
+        if(!figureRef.current) return;
+
```

Comment:
Remove the whitespace here (the line return). It'll help readability of the code.